refactor: remove `ContractCallError`

jakmeier opened the pull request:
The indirection from `FunctionCallError` to `ContractCallError`
to `FunctionCallErrorSer` seems to be obsolete after the latest
rounds of refactoring. Let's remove it.

jakmeier commented:
another one for #7818 

@matklad I am not sure if we have any reasons to care about preserving what exact `FunctionCallErrorSer` we produce. If not, I think we can avoid converting errors to stings here:

https://github.com/near/nearcore/pull/7881/files#diff-20827cd3bea08c28cc7c00c44e31d4a416dc2028261161a63491149f7f25994aR303-R316

matklad commented:
>If not, I think we can avoid converting errors to stings here:

Conversion to strings is somewhat intentional: the *structured* errors are essentially legacy errors: we don't want to tie our DB schema to the shape of random error types in Rust. Ideally, what we store in db is

```rust
struct Error {
  code: u16,
  message: String,
}
```


The current `FunctionCallSer` is not quite that yet, but that's where the idea of stringifying comes from.

matklad commented on runtime/near-vm-errors/src/lib.rs:
```diff
@@ -298,6 +297,27 @@ pub enum InconsistentStateError {
+    fn from(outer_err: FunctionCallError) -> Self {
+        match outer_err {
+            FunctionCallError::CompilationError(e) => FunctionCallErrorSer::CompilationError(e),
+            FunctionCallError::MethodResolveError(e) => FunctionCallErrorSer::MethodResolveError(e),
+            // TODO: consider using FunctionCallErrorSer::HostError
+            FunctionCallError::HostError(ref _e) => {
+                FunctionCallErrorSer::ExecutionError(outer_err.to_string())
+            }
+            // TODO: consider using FunctionCallErrorSer::LinkError
+            FunctionCallError::LinkError { msg } => {
+                FunctionCallErrorSer::ExecutionError(format!("Link Error: {}", msg))
+            }
+            // TODO: consider using FunctionCallErrorSer::WasmTrap
+            FunctionCallError::WasmTrap(ref _e) => {
+                FunctionCallErrorSer::ExecutionError(outer_err.to_string())
+            }
```
Yeah, so this is not TODO, but rather intentional design. The "structured" varians for FunctionCallErrorSer are only there to deserialize legacy error, all new data shoudl not depend on details of representation of errors.

matklad reviewed (approved):
LGTM with TODOs replaced with "yes, this is how it should look"

jakmeier commented:
> > If not, I think we can avoid converting errors to stings here:
> 
> Conversion to strings is somewhat intentional: the _structured_ errors are essentially legacy errors: we don't want to tie our DB schema to the shape of random error types in Rust. Ideally, what we store in db is
> 
> ```rust
> struct Error {
>   code: u16,
>   message: String,
> }
> ```
> 
> The current `FunctionCallSer` is not quite that yet, but that's where the idea of stringifying comes from.

Ah, I see. That makes sense now!

I removed the TODOs and added some comments that this is deliberate and that we shouldn't use the old errors for new data.
